products/jbrowse-cli/src/commands/text-index.ts

```diff
@@ -0,0 +1,592 @@
+    '$ jbrowse text-index ... --out=path_to_configuration_file',
+  ]
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+    tracks: flags.string({
+      description: `Specify the tracks to index, formatted as comma separated trackIds`,
+    }),
+    individual: flags.boolean({
+      description: `Only make a single-track text index for the given track`,
+    }),
+    location: flags.string({
+      description: `Establish a location for the output files`,
+    }),
+    target: flags.string({
+      description:
+        'Path to config file in JB2 installation directory to read from.',
+    }),
+    out: flags.string({
+      description: 'Synonym for target',
+    }),
+  }
+
+  // Called when running the terminal command. Parses the given flags
+  // and tracks associated. Gets their information and sends it to the
+  // appropriate file parser to be indexed
+  async run() {
+    const { flags: runFlags } = this.parse(TextIndex)
+
+    const configPath: string = path.join(__dirname, '..', '..', 'config.json')
+    const output = runFlags?.target || runFlags?.out || configPath || '.'
+    const isDir = (await promises.lstat(output)).isDirectory()
+    this.target = isDir ? `${output}/config.json` : output
+    let fileDirectory: string = path.join(__dirname)
+
+    if (runFlags.individual) {
+      if (runFlags.tracks) {
+        if (runFlags.location) {
+          fileDirectory = runFlags.location
+        }
+
+        const trackIds: Array<string> = runFlags.tracks?.split(',')
+        if (trackIds.length > 1) {
+          this.error(
+            'Error, --individual flag only allows one track to be indexed',
+          )
+        } else {
+          const indexConfig = await this.getIndexingConfigurations(
+            trackIds,
+            this.target,
+          )
+          const indexAttributes = indexConfig[0]?.attributes || []
+          const uri =
+            indexConfig[0].indexingConfiguration?.gffLocation.uri || ''
+
+          await this.indexDriver(
+            uri,
+            false,
+            indexAttributes,
+            fileDirectory,
+          ).catch(err => this.error(err))
+        }
+      } else {
+        this.error('Error, please specify a track to index.')
+      }
+    } else if (runFlags.tracks) {
+      if (runFlags.location) {
+        fileDirectory = runFlags.location
+      }
+
+      const trackIds: Array<string> = runFlags.tracks.split(',')
+      const uris: Array<string> = []
+      const indexConfig = await this.getIndexingConfigurations(
+        trackIds,
+        this.target,
+      )
+
+      for (const x in indexConfig) {
+        uris.push(indexConfig[x]?.indexingConfiguration?.gffLocation.uri || '')
+      }
+      const indexAttributes = indexConfig[0]?.attributes || []
+
+      await this.indexDriver(
+        uris,
+        false,
+        indexAttributes,
+        fileDirectory,
+      ).catch(err => this.error(err))
```

Comment:
I think .catch should not be needed here because as far as i know, the oclif framework has a default catch handler that does essentially the same thing